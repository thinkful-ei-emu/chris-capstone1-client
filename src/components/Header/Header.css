.Header {
    height: 7vh;
    display: flex;
    align-items: normal;
    justify-content: space-between;
  }

.Header_image {
  height: 30px;
  margin-top: 15px;
  padding-left: 10px;
}
  
  .center-nav {
    padding-top: 20px;
    margin-left: 20px;
  }
  
  .Header a {
    color: #444;
    text-decoration: none;
  }
  
  .Header__tagline--wide {
    opacity: 0.35;
    margin-left: 10px;
  }

  .Header__tagline--narrow {
    opacity: 0.35;
    display: block;
    text-align: center;
    display: none;
    padding-top: 10px;
  }

  .Header__logged-in {
    padding-right: 10px;
    padding-top: 20px;
  }
  
  .Header__not-logged-in {
    display: flex;
    flex: 1;
    align-items: center;
    justify-content: flex-end;
    padding-right: 10px;
    padding-top: 30px;
  }
  
  .Header__not-logged-in a:last-child {
    background: #FF8D2E;
    color: white;
    padding: 5px 15px;
    border-radius: 15px;
    margin-left: 10px;
  }
  
  .Header__logged-in a, .center-nav a {
    border: 1px solid rgba(0, 0, 0, 0.25);
    padding: 5px 15px;
    border-radius: 15px;
  }
  
  @media screen and (min-width: 490px) {
    
    .Header_image {
      height: 55px;
    }

    .center-nav {
      padding-top: 35px;
      margin-right: 65px;
    }
    .Header__logged-in {
      padding-top: 35px;
    }
    
  }
  
  @media screen and (max-width: 490px) {
    .Header__tagline--wide {
      display: none;
    }
  
    .Header__tagline--narrow {
      display: inherit;
    }
  }
  