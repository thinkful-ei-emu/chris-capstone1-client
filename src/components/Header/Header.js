import React from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import './Header.css';

export default class Header extends React.Component {
    handleLogoutClick = () => {
        TokenService.clearAuthToken();
    }

    renderLogoutLink() {
        return (
            <nav className='Nav'>
                <div className='left-nav'>
                    <Link to='/home'>
                        Home
                    </Link>
                    <Link to='/userhome'>
                        Your BookShelf
                    </Link>
                </div>
                <div className='navLoggedIn'>
                    <Link
                    onClick={this.handleLogoutClick}
                    to='/'>
                        Logout
                    </Link>
                </div>
            </nav>
        )
    }

    renderLoginLink() {
        return (
            <nav className='Nav'>
                <div className='left-nav'>
                    <Link to='/'>
                        Home
                    </Link>
                </div>
                <div className='navNotLoggedIn'>
                    <Link
                        to='/login'>
                            Log In
                    </Link>
                    <Link
                        to='/register'>
                            Register
                    </Link>
                </div>
            </nav>
        )
    }

    render() {
        return (
        <>
            {TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}
        </>
        )
    }
}