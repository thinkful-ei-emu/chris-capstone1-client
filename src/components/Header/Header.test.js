import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

describe('Header component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
          <BrowserRouter>
            <Header loggedIn={true} logoutSuccess={() => {}} />
          </BrowserRouter>,
          div
        )
        ReactDom.unmountComponentAtNode(div)
      })
    it('renderes Header by default', () => {
        expect(toJson(shallow(<Header loggedIn={true} logoutSuccess={() => {}} />))).toMatchSnapshot() 
    })
})