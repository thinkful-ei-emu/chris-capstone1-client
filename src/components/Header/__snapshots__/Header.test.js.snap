// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renderes Header by default 1`] = `
<Fragment>
  <nav
    className="Header"
  >
    <Link
      to="/"
    >
      <img
        alt="the mobile bookshelf logo"
        className="Header_image"
        src="MB_Logo.jpg"
      />
    </Link>
    <div
      className="center-nav"
    >
      <Link
        to="/yourshelf"
      >
        Your BookShelf
      </Link>
    </div>
    <div
      className="Header__logged-in"
    >
      <Link
        onClick={[Function]}
        to="/"
      >
        Logout
      </Link>
    </div>
  </nav>
</Fragment>
`;
